site/static/test-gallery.ts

```diff
@@ -0,0 +1,58 @@
+d3.select('#vl-version').text(vl.version);
+d3.select('#vg-version').text(vg.version);
+
+// parse url parameters
+var params = function() {
+  var query = location.search.slice(1);
+  if (query.slice(-1) === '/') {
+    query = query.slice(0,-1);
+  }
+  return query
+    .split('&')
+    .map(function(x) { return x.split('='); })
+    .reduce(function(a, b) {
+      a[b[0]] = b[1]; return a;
+    }, {});
+};
+
+d3.json('examples/all-examples.json', function(examples) {
+  console.log(examples);
+
+  render();
+
+  function render() {
+    var viz = d3.select('div.viz-list').selectAll('.viz').data(examples);
+
+    viz.exit().remove();
+
+    var vizEnter = viz
+      .enter()
+      .append('div')
+      .attr('class', 'viz')
+      .attr('id', function(d:string) { return d; });
+
+    vizEnter.append('h3').text(function(d:string) { return d; });
+    vizEnter.append('div').attr('class', 'desc');
+    vizEnter.append('div').attr('class', 'view');
+
+    examples.forEach(function(example) {
+      d3.json('examples/specs/' + example + '.json', function(error, vlSpec) {
```

Comment:
Can you add the name of the file somewhere visible? We often need to know which example is broken and it's hard to find it.